api/src/main/java/com/datastrato/gravitino/file/Fileset.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.file;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.Namespace;
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * An interface representing a fileset under a schema {@link Namespace}. A fileset is a virtual
+ * concept of the file or directory that is managed by Gravitino. Users can create a fileset object
+ * to manage the non-tabular data on the FS-like storage. The typical use case is to manage the
+ * training data for AI workloads. The major difference compare to the relational table is that the
+ * fileset is schema-free, the main property of the fileset is the storage location of the
+ * underlying data.
+ *
+ * <p>{@link Fileset} defines the basic properties of a fileset object. A catalog implementation
+ * with {@link FilesetCatalog} should implement this interface.
+ */
+public interface Fileset extends Auditable {
+
+  /** An enum representing the type of the fileset object. */
+  enum Type {
+    MANAGED, // Fileset is managed by Gravitino. When specified, the data will be deleted when the
+    // fileset object is deleted.
+    EXTERNAL // Fileset is not managed by Gravitino. When specified, the data will not be deleted
+    // when the fileset object is deleted.
+  }
+
+  /** @return Name of the fileset object. */
+  String name();
+
+  /** @return The comment of the fileset object. Null is returned if no comment is set. */
+  @Nullable
+  default String comment() {
+    return null;
+  }
+
+  /** @return The type of the fileset object. */
+  Type type();
+
+  /**
+   * Get the storage location of the file or directory path that is managed by this fileset object.
+   *
+   * @return The storage location of the fileset object.
+   */
+  String storageLocation();
```

Comment:
Does need a virtual path property here? I saw that there is a virtual path in the volume of Databricks, and resources can be accessed through this virtual path:  
![image](https://github.com/datastrato/gravitino/assets/26177232/150e7e2f-430c-4245-928d-ae1a519b16af)

Reply:
> Does need a virtual path property here? I saw that there is a virtual path in the volume of Databricks, and resources can be accessed through this virtual path: ![image](https://private-user-images.githubusercontent.com/26177232/296910747-150e7e2f-430c-4245-928d-ae1a519b16af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzNzEzNDQsIm5iZiI6MTcwNTM3MTA0NCwicGF0aCI6Ii8yNjE3NzIzMi8yOTY5MTA3NDctMTUwZTdlMmYtNDMwYy00MjQ1LTkyOGQtYWUxYTUxOWIxNmFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDAyMTA0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyNDMxZDRjYjJmYjVjZDcxM2YxM2U0OWVkOGY4MzRjODlmZmU5ZTQ0M2Y1Y2Y2Nzg0OWYyMWI5NmM2ZTQ3MmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RUGh5T6vgZycSE1xXIVlJykkuNeV9_-qlWllRg9L0eE)

After thinking about it again, if the conversion from virtual path to metalake.catalog.schema.fileset can be realized in Proxy FS, it seems that it is not necessary.

Reply:
Virtual path is generated from NameIdentifier, which is "metalake.catalog.schema.fileset", so the virtual path will be "/metadata/catalog/schema/fileset".